tests/measurements/test_shots.py

```diff
@@ -130,6 +125,21 @@ def test_repr(self, expected_str, shots_obj):
         """Test that the repr is correct"""
         assert expected_str == repr(shots_obj)
 
+    def test_eq(self):
+        """Test that the equality function behaves correctly"""
+        for s in self.shot_data:
+            assert s == copy.copy(s)
```

Comment:
Might not actually be a good test, because `copy(s)` and `deepcopy(s)` just give back the original instance. This is because `Shots` is immutable. We don't need to worry about changing `s` and having it modify its copy.

Just like how:
```
>>> a = (1,2)
>>> copy.deepcopy(a) is a
True
```

Reply:
Maybe instead ` s == Shots(s.shot_copies if s.shot_copies else None)`
This forces the construction of an identical object that's located somewhere else in memory.

Reply:
Adde 👍🏼